client/src/pages/Home/index.js

```diff
@@ -1,24 +1,65 @@
-import { useEffect, useState } from "react";
+import React, { useEffect, useState } from "react";
 
 import { Pagination, ResourceList } from "../../components";
 import { useSearchParams } from "../../hooks";
 import { ResourceService, useService } from "../../services";
 
-export function Home() {
+const Home = () => {
 	const resourceService = useService(ResourceService);
 	const searchParams = useSearchParams();
-	const [{ lastPage, resources } = {}, setEnvelope] = useState();
+	const [{ lastPage, resources } = {}, setEnvelope] = useState({});
+
+	const storedPageSize = localStorage.getItem("pageSizePreference");
+	let initialPageSize;
+
+	try {
+		initialPageSize = storedPageSize ? JSON.parse(storedPageSize) : "default";
+	} catch (error) {
+		// Handle the error as needed
+		initialPageSize = "default";
+		throw new Error("Error parsing page size preference:", error);
+	}
+
+	const [pageSize, setPageSize] = useState(initialPageSize);
+
+	// This function is to save the selected page-size into local storage
+	const savePageSizePreference = (pageSize) => {
+		localStorage.setItem("pageSizePreference", JSON.stringify(pageSize));
+	};
 
 	useEffect(() => {
-		resourceService.getPublished(searchParams).then(setEnvelope);
-	}, [resourceService, searchParams]);
+		// Using the selected page size in my API request
+		resourceService
+			.getPublished({ ...searchParams, pageSize })
+			.then(setEnvelope)
+			.catch((error) => {
+				// Handle the error from the API request as needed
+				throw new Error("Error fetching published resources:", error);
+			});
+	}, [resourceService, searchParams, pageSize]);
 
 	return (
 		<section>
-			<ResourceList resources={resources ?? []} />
-			<Pagination lastPage={lastPage ?? 1} />
+			{/* Create the UI for page size selection */}
+			<label htmlFor="pageSize">Page Size: </label>
+			<select
+				id="pageSize"
+				value={pageSize}
+				onChange={(e) => {
+					const selectedPageSize = e.target.value;
+					setPageSize(selectedPageSize);
+					savePageSizePreference(selectedPageSize);
+				}}
+			>
+				<option value="default">Default</option>
+				<option value="small">Small</option>
+				<option value="large">Large</option>
```

Comment:
❓ What are these values supposed to represent? Assuming they follow what's already on the page, i.e.:

- Default: 20
- Small: 10
- Large: 50

it would be helpful to include those values, and this is an odd order to put them in.